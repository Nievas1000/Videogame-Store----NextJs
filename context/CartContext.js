import { createContext, useContext, useState } from "react";

export const GameContext = createContext()

export const useGame = () => {return useContext(GameContext)}

export const CartProvider = ({children}) =>{
    const [carro, setCarro] = useState([])
    const [cantidad, setCantidad] = useState(0)
    const [carroVisible, setVisibilidad] = useState(false)


    const agregarAlCarro = (game) =>{
        if(carro.find(x => x.id === game.id)){
            return alert("You already have this game in your cart")
        }
        setCarro([
            ...carro,
            game,
        ])
        setCantidad(cantidad + 1)
    }
    const mostrarCarro = () =>{
        setVisibilidad(!carroVisible)
    }

    const removeFromCart = (game) =>{
        let indice = carro.indexOf(carro.find(x => x.id === game.id))
        carro.splice(indice,1)
        setCarro([...carro])
        setCantidad(cantidad - 1)
    }

    return(
        <GameContext.Provider value={{carro, cantidad, carroVisible, agregarAlCarro, mostrarCarro, removeFromCart}}>{children}</GameContext.Provider>
    )
}