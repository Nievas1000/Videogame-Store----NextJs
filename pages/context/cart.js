import { createContext, useContext, useState } from "react";

export const GameContext = createContext()

export const useGame = () => {return useContext(GameContext)}

export const CartProvider = ({children}) =>{
    const [carro, setCarro] = useState([])
    const [cantidad, setCantidad] = useState(0)
    const [carroVisible, setVisibilidad] = useState(false)


    const agregarAlCarro = (game) =>{
        if(carro.find(x => x.id === game.id)){
            return alert("You already have this game in your cart")
        }
        setCarro([
            ...carro,
            game,
        ])
        setCantidad(cantidad + 1)
    }
    const mostrarCarro = () =>{
        setVisibilidad(!carroVisible)
    }

    return(
        <GameContext.Provider value={{carro, cantidad, carroVisible, agregarAlCarro, mostrarCarro}}>{children}</GameContext.Provider>
    )
}