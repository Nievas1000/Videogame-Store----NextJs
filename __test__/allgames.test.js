import { render } from "@testing-library/react";
import Games, {getStaticProps} from "../pages/allgames"
import {useGame} from "../pages/context/CartContext"

jest.mock("../pages/context/CartContext")

describe("All games", () =>{

    it("render games", () => {
        const {games} = render(<Games games={[]}/>)
        expect(games).toMatchSnapshot()
    })

    describe("getStaticProps", () => {
        it("return allgames", async () =>{
            global.fetch = jest.fn()
            .mockImplementation(url => {
                return new Promise(resolve => {
                    resolve({
                        json: () => Promise.resolve({
                            slice: () => "allgames"
                        })
                    })
                })
            })
            const {props} = await getStaticProps()
            expect(props.games).toBe("allgames")
        })
    })

    beforeEach(() => {
        const datos = []
        useGame.mockImplementation(() => datos)
      })
})