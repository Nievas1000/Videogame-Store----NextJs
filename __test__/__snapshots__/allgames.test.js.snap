// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All games render games 1`] = `undefined`;
